src/source/PeerConnection/Rtcp.c

```diff
@@ -139,6 +139,138 @@ static STATUS onRtcpReceiverReport(PRtcpPacket pRtcpPacket, PKvsPeerConnection p
+    packetStatusCount = TWCC_PACKET_STATUS_COUNT(pRtcpPacket->payload);
+
+    packetsRemaining = packetStatusCount;
+    chunkOffset = 16;
+    while (packetsRemaining > 0 && chunkOffset < pRtcpPacket->payloadLength) {
+        packetChunk = getUnalignedInt16BigEndian(pRtcpPacket->payload + chunkOffset);
+        if (IS_TWCC_RUNLEN(packetChunk)) {
+            packetsRemaining -= TWCC_RUNLEN_GET(packetChunk);
+        } else {
+            packetsRemaining -= MIN(TWCC_STATUSVECTOR_COUNT(packetChunk), packetsRemaining);
+        }
+        chunkOffset += 2;
+    }
+
+    recvOffset = chunkOffset;
+    chunkOffset = 16;
+    packetSeqNum = baseSeqNum;
+    packetsRemaining = packetStatusCount;
+    while (packetsRemaining > 0) {
+        packetChunk = getUnalignedInt16BigEndian(pRtcpPacket->payload + chunkOffset);
+        statusSymbol = TWCC_RUNLEN_STATUS_SYMBOL(packetChunk);
+        if (IS_TWCC_RUNLEN(packetChunk)) {
+            for (i = 0; i < TWCC_RUNLEN_GET(packetChunk); i++) {
+                recvDelta = MIN_INT16;
+                if (TWCC_IS_SMALLDELTA(statusSymbol)) {
+                    recvDelta = (INT16) pRtcpPacket->payload[recvOffset];
+                    recvOffset++;
+                } else if (TWCC_IS_LARGEDELTA(statusSymbol)) {
+                    recvDelta = getUnalignedInt16BigEndian(pRtcpPacket->payload + recvOffset);
+                    recvOffset += 2;
+                } else if (TWCC_IS_NOTRECEIVED(statusSymbol)) {
+                    DLOGV("runLength packetSeqNum %u not received", packetSeqNum);
+                    if (pKvsPeerConnection->onPacketNotReceived != NULL) {
+                        pKvsPeerConnection->onPacketNotReceived(pKvsPeerConnection->onPacketNotReceivedCustomData, packetSeqNum);
+                    }
+                } else {
+                    DLOGD("runLength unhandled statusSymbol %u", statusSymbol);
+                }
+                if (recvDelta != MIN_INT16) {
+                    DLOGV("runLength packetSeqNum %u recvDelta %d usec", packetSeqNum,
+                          KVS_CONVERT_TIMESCALE(recvDelta, TWCC_TICKS_PER_SECOND, MICROSECONDS_PER_SECOND));
+                    if (pKvsPeerConnection->onPacketReceived != NULL) {
+                        pKvsPeerConnection->onPacketReceived(pKvsPeerConnection->onPacketReceivedCustomData, packetSeqNum,
+                                                             KVS_CONVERT_TIMESCALE(recvDelta, TWCC_TICKS_PER_SECOND, MICROSECONDS_PER_SECOND));
+                    }
+                }
+                packetSeqNum++;
+                packetsRemaining--;
+            }
+        } else {
+            statuses = MIN(TWCC_STATUSVECTOR_COUNT(packetChunk), packetsRemaining);
+
+            for (i = 0; i < statuses; i++) {
+                statusSymbol = TWCC_STATUSVECTOR_STATUS(packetChunk, i);
+                recvDelta = MIN_INT16;
+                if (TWCC_IS_SMALLDELTA(statusSymbol)) {
```

Comment:
We can convert this entire if...else block into switch case if we use enums instead of #defs. 

```
switch(statusSymbol) {
    case TWCC_IS_SMALLDELTA:
       recvDelta = (INT16) pRtcpPacket->payload[recvOffset];
       recvOffset++;
       break;
    ....
    default:
      DLOGD("statusVector unhandled statusSymbol %u", statusSymbol);
}
```

Reply:
Reference: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/987/files#r534582509

Reply:
cool